bpf/flows.c

```diff
@@ -55,16 +52,53 @@
  */
 #include "pkt_translation.h"
 
-static inline void update_existing_flow(flow_metrics *aggregate_flow, pkt_info *pkt, u64 len,
-                                        u32 sampling) {
+// return 0 on success, 1 if capacity reached
+static __always_inline int add_observed_intf(flow_metrics *value, pkt_info *pkt, u32 if_index,
+                                             u8 direction) {
+    if (value->nb_observed_intf >= MAX_OBSERVED_INTERFACES) {
+        BPF_PRINTK("observed interface missed (array capacity reached) for ifindex %d\n", if_index);
+        return 1;
+    }
+    for (u8 i = 0; i < value->nb_observed_intf; i++) {
+        if (value->observed_intf[i] == if_index) {
+            if (value->observed_direction[i] != direction &&
+                value->observed_direction[i] != OBSERVED_DIRECTION_BOTH) {
+                // Same interface seen on a different direction => mark as both directions
+                value->observed_direction[i] = OBSERVED_DIRECTION_BOTH;
```

Comment:
won't this case issues in flp generating the other flp direction field ?

Reply:
FLP doesn't use the interface direction to figure out the node direction, it does that only by comparing agent IP versus src/dest node IP
I don't think there's anything that reads the IfDirections field, except in the console for display...

Reply:
what is the impact of using direction both when we filter flows on direction ?